### ISSUE TITLE
Excessive AL02 on SELECT * REPLACE bigquery

### ISSUE THREAD

**Issue by Lauriane96** [sqlfluff/sqlfluff#6775]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Consider the following test.sql file before applied sqlfluff linter :

```sql
CREATE OR REPLACE TABLE my_db.test (
    id STRING,
    name_test STRING
);

INSERT INTO my_db.test
VALUES ("1", "Mystic Black"),
("2", "Mini green");

select * replace (concat(name, "_new") AS name_test)
FROM my_db.test;
``` 

Invoking linting under version 3.3.1

```shell
sqlfluff lint --dialect bigquery test.sql
``` 

Result in error on AL02 however the detected alias is mandatory in REPLACE statement:
```shell
FAIL                                                                                                               
L:  10 | P:   1 | AM04 | Query produces an unknown number of result columns.
                       | [ambiguous.column_count]
L:  10 | P:  45 | AL02 | Implicit/explicit aliasing of columns.
                       | [aliasing.column]
All Finished 📜 🎉!
``` 

So after formatting the sql file, it removed the mandatory 'AS' in REPLACE statement: 
```sql
CREATE OR REPLACE TABLE my_db.test (
    id STRING,
    name_test STRING
);

INSERT INTO my_db.test
VALUES ("1", "Mystic Black"),
("2", "Mini green");

SELECT * REPLACE (CONCAT(name_test, "_new") name_test)
FROM my_db.test;
``` 

### Expected Behaviour

The only thing that have to be updated might be about the uppercase set in sqlfluff params. The 'AS' keyword must be kept since it is mandatory inside SELECT statement with REPLACE ( https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#select_replace ):

```sql
CREATE OR REPLACE TABLE my_db.test (
    id STRING,
    name_test STRING
);

INSERT INTO my_db.test
VALUES ("1", "Mystic Black"),
("2", "Mini green");

SELECT * REPLACE (CONCAT(name_test, "_new") AS name_test)
FROM my_db.test;
``` 

Invoking linting under version 3.3.1

```shell
sqlfluff lint --dialect bigquery test.sql
``` 

Result in error on AL02 since the detected alias is mandatory:
```shell
FAIL                                                                                                               
L:  10 | P:   1 | AM04 | Query produces an unknown number of result columns.
                       | [ambiguous.column_count]
All Finished 📜 🎉!
``` 

### Observed Behaviour

However, the 'AS' keyword inside the REPLACE statement is removed after formatting using SQLFLUF, so the query execution is KO after formatting.

### How to reproduce

```sql
CREATE OR REPLACE TABLE my_db.test (
    id STRING,
    name_test STRING
);

INSERT INTO my_db.test
VALUES ("1", "Mystic Black"),
("2", "Mini green");

select * replace (concat(name, "_new") AS name_test)
FROM my_db.test;
``` 

Invoking linting under version 3.3.1

```shell
sqlfluff lint --dialect bigquery test.sql
``` 

### Dialect

bigquery

### Version

sqlfluff version = 3.3.1

### Configuration

[sqlfluff]
dialect = bigquery
templater = jinja
sql_file_exts = .sql

max_line_length = 140

exclude_rules = convention.select_trailing_comma, ambiguous.join, layout.end_of_file, references.from

[sqlfluff:rules:aliasing.table]
aliasing = implicit

[sqlfluff:rules:aliasing.column]
aliasing = implicit

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
